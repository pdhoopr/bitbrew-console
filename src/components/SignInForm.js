import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import SignIn from '../models/SignIn';
import { RaisedButton } from './Buttons';
import { FlexEnd } from './Flexboxes';
import { Input } from './Inputs';

const Wrapper = styled.form`
  background-color: var(--color-white);
  border-radius: var(--corner-radius);
  box-shadow: var(--elevation-low);
  margin-left: auto;
  margin-right: auto;
  max-width: var(--size-320);
  padding: var(--size-16) var(--size-24);
`;

const Title = styled.h1`
  font-size: var(--size-20);
  font-weight: var(--weight-normal);
  line-height: var(--size-28);
  margin-bottom: var(--size-16);
  margin-top: 0;
`;

const Label = styled.label`
  display: block;
  font-weight: var(--weight-bold);
  margin-bottom: var(--size-16);

  & ${/* sc-selector */ Input} {
    margin-top: var(--size-8);
  }
`;

class SignInForm extends React.Component {
  values = SignIn.create({}, { onSubmit: this.props.signIn });

  render() {
    return (
      <Wrapper
        onSubmit={this.values.submit}
        noValidate
        data-testid="sign-in-form"
      >
        <Title>Sign In</Title>
        <Label htmlFor="accessToken">
          <span>Access Token</span>
          <Input
            id="accessToken"
            value={this.values.accessToken}
            onChange={this.values.update}
            type="text"
          />
        </Label>
        <FlexEnd>
          <RaisedButton>Sign in</RaisedButton>
        </FlexEnd>
      </Wrapper>
    );
  }
}

SignInForm.propTypes = {
  signIn: PropTypes.func.isRequired,
};

export default observer(SignInForm);
