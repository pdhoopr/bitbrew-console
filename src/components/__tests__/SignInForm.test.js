import { shallow } from 'enzyme';
import React from 'react';
import SignInForm from '../SignInForm';

test('renders DOM for the sign in form', () => {
  const dom = shallow(<SignInForm signIn={jest.fn()} />);
  expect(dom).toMatchSnapshot();
});
