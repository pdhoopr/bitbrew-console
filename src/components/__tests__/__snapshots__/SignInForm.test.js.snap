// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DOM for the sign in form 1`] = `
<styled.form
  data-testid="sign-in-form"
  noValidate={true}
  onSubmit={[Function]}
>
  <styled.h1>
    Sign In
  </styled.h1>
  <styled.label
    htmlFor="accessToken"
  >
    <span>
      Access Token
    </span>
    <styled.input
      id="accessToken"
      onChange={[Function]}
      type="text"
      value=""
    />
  </styled.label>
  <styled.div>
    <styled.button>
      Sign in
    </styled.button>
  </styled.div>
</styled.form>
`;
